src/librustc_mir/transform/promote_consts.rs

```diff
@@ -502,9 +502,33 @@ impl<'tcx> Validator<'_, 'tcx> {
     fn validate_place(&self, place: PlaceRef<'tcx>) -> Result<(), Unpromotable> {
         match place {
             PlaceRef { local, projection: [] } => self.validate_local(local),
-            PlaceRef { local: _, projection: [proj_base @ .., elem] } => {
+            PlaceRef { local, projection: [proj_base @ .., elem] } => {
                 match *elem {
-                    ProjectionElem::Deref | ProjectionElem::Downcast(..) => {
+                    ProjectionElem::Deref => {
+                        let mut not_promotable = true;
+                        if let TempState::Defined { location, .. } = self.temps[local] {
```

Comment:
leave a comment here stating that dereferencing a reference to a static this way can only get generated from a direct access which is essentially `*&STATIC`, so we should just allow it.